foreman/dev_managed_dhcp/README.md

```diff
@@ -0,0 +1,191 @@
+# Foreman - Baremetal DEV environment provisioning
+
+Local development setup for simulating bare-metal provisioning in environments where Foreman is not directly accessible from the machines, but only via Smart Proxy.
+
+## Preconditions
+* Running foreman DEV setup
+* Running Smart Proxy DEV setup
+* [Installed virtualization](https://docs.fedoraproject.org/en-US/quick-docs/virtualization-getting-started/)
+* (optional) [HTTPS & DEV ENV](https://github.com/stejskalleos/knowledge_base/blob/main/foreman/https_dev_env.md)
+
+**Tools**
+
+* Libvirt for virtual machines (our 'baremetals') & subnet configuration
+* [ISC DHCP](https://www.isc.org/dhcp/)
+* DNS
+
+**Smart Proxy features**
+
+* DHCP
+* DNS
+* TFTP
+* Registration
+* Facts
+* Templates
+
+**Data**
+* Domain: `foreman-isc.lan`
+* Subnet: `192.168.66.1`
+* Provisioning interface: `virbr66`
+
+## Libvirt subnet
+See the `libvirt/foreman-isc.lan.xml` for the network configuration.
+
+```
+virsh net-create ./foreman-isc.lan.xml
+virsh net-define ./foreman-isc.lan.xml
+virsh net-autostart foreman-isc.lan
+
+virsh net-info foreman-isc.lan
+```
+
+## Firewall
+
+```shell
+firewall-cmd --change-interface virbr66 --zone trusted
+firewall-cmd --change-interface virbr66 --zone trusted --permanent
+nmcli connection modify virbr66 connection.zone trusted
+```
+
+## TFTP
+
+```shell
+dnf install -y syslinux
+mkdir -p /var/lib/tftpboot
+mkdir -p /var/lib/tftpboot/{boot,grub,grub2,pxelinux.cfg}
+
+cp /usr/share/syslinux/{pxelinux.0,menu.c32,chain.c32,ldlinux.c32,libcom32.c32,libutil.c32} /var/lib/tftpboot
+
+# Change file owner to your account
+chown your-account:your-group -R /var/lib/tftpboot
+```
+
+## DHCP
+
+### Installation
+
+```
+dnf install dhcp-server
+```
+
+### Configuration
+
+Configure the `/etc/dhcp/dhcpd.conf`
+See the `dhcp/dhcpd.conf` file.
+
+### Service
+
+```
+cp /usr/lib/systemd/system/dhcpd.service /etc/systemd/system/dhcpd.service
+```
+
+Find the line starting with `ExecStart= `and add the interface name at the end of this line. It should look like this:
+
+```
+# /etc/systemd/system/dhcpd.service
+
+...
+
+[Service]
+ExecStart=/usr/sbin/dhcpd -f -cf /etc/dhcp/dhcpd.conf virbr66 -user dhcpd -group dhcpd --no-pid $DHCPDARGS
+
+...
```

Comment:
This is not how you do things with systemd. Instead you create drop in files:
````suggestion
systemctl edit dhcpd
```

Add the following content:
```ini
[Service]
ExecStart=
ExecStart=/usr/sbin/dhcpd -f -cf /etc/dhcp/dhcpd.conf virbr66 -user dhcpd -group dhcpd --no-pid $DHCPDARGS
```
````

Reply:
That's also what we do in our installer: we create `/etc/systemd/system/dhcpd.service.d/interfaces.conf` here:

https://github.com/theforeman/puppet-dhcp/blob/fa9e69407666f080bafb87008551fb85c2337cdd/manifests/init.pp#L104-L109

And that renders the following template:

https://github.com/theforeman/puppet-dhcp/blob/master/templates/redhat/systemd-dropin.conf.erb